### DISCUSSION TITLE
Replace `url` dependency with modern URL API?

### DISCUSSION THREAD

**Discussion by MuTsunTsai** [pixijs/pixijs#9413] [Ideas]
@pixi/utils has a dependency on `url`. As far as I can tell, it is essentially used in only one place:
https://github.com/pixijs/pixijs/blob/dev/packages/utils/src/network/determineCrossOrigin.ts
and apparently this file hasn't been touched for years (there's even a comment about IE9 which should be long dropped by Pixi). I believe that the `url` dependency and the `tempAnchor` stuff can all be replaced by simply
```
const parsedUrl = new URL(url, document.baseURI);
```
This works in all supposedly targeted browsers, and can save us a few dependencies to bundle and the annoying "querystring deprecated" warning.

The only thing I hesitate about making a PR is that @pixi/utils actually exports `url` (in https://github.com/pixijs/pixijs/blob/dev/packages/utils/src/index.ts), and I'm not entirely sure if it is used elsewhere in the ecosystem. What do you guys think?

**Comment by SuperSodaSea** [comment]
I think it would be great to replace package `url` with native `URL`. For compatibility, we can mark `url` as deprecated and completely remove it completely later in PixiJS v8. Feel free to make a PR!

#### packages/utils/src/index.ts

```diff
  /**
   * Node.js compatible URL utilities.
   * @see https://www.npmjs.com/package/url
   * @memberof PIXI.utils
   * @name url
   * @member {object}
+  * @deprecated since 7.3.0
   */
  export * from './url';
```

#### packages/utils/src/url.ts

```ts
export const url = {
    /**
     * @deprecated since 7.3.0
     */
    parse: {
        get()
        {
            // #if _DEBUG
            deprecation('7.3.0', 'utils.url.parse is deprecated, use native URL API instead');
            // #endif

            return _parse as ParseFunction;
        },
    },
    // And format & resolve
};
```

**Reply by MuTsunTsai** [reply]
Alright then, PR made.
